### PR TITLE
Update addin and module best practices for Cake 2.0

### LINKED ISSUE #2222: Update extension best practices for Cake 2.0
Based on https://github.com/cake-build/cake/discussions/3575 best practices for addins and modules for Cake 2.0 will be to target `netcoreapp3.1`, `net5.0`, `net6.0`. 

This needs to be updated in [Best practices for writing addins](https://cakebuild.net/docs/extending/addins/best-practices) and [Best practices for writing modules](https://cakebuild.net/docs/extending/modules/best-practices).

### PR DESCRIPTION
**Pull request by pascalberger**
Update addin and module best practices for Cake 2.0

Fixes #2222

### CODE REVIEW

File: input/docs/extending/modules/best-practices.md [hunk 1]
```diff
@@ -102,12 +102,11 @@ The wording of each guideline indicates how strong the recommendation is:
 
 ## .NET target version
 
-**_§2.5_** **Do** target `netstandard2.0`.
+**_§2.5_** **Do** target `netcoreapp3.1`.
```
Comment (pascalberger) [inline input/docs/extending/modules/best-practices.md:105]: Is targeting only `netcoreapp3.1` what we suggest due to the issues with multi-targeting?

**Review by nils-a — APPROVED** [review]
LGTM!

File: input/docs/extending/modules/best-practices.md (hunk 1 above)
Comment (augustoproiete) [reply input/docs/extending/modules/best-practices.md:105]: Yes, that seems to be the right choice until we drop support for `netcoreapp3.1`.

**Review by augustoproiete — APPROVED** [review]
LGTM!

**Pull request merged**